### PR TITLE
Docs of CamelCase plural/singular limitations.

### PR DESCRIPTION
**Pull request by dereuromark**
If performance is an issue, we could use this as alternative to https://github.com/cakephp/cakephp/pull/14667 and stop using/promoting the faulty CamelCase versions.

Also added 4.x version of inflection demo tool (live).

### CODE REVIEW

File: en/core-libraries/inflector.rst [hunk 1]
```diff
@@ -72,20 +69,24 @@ to support other languages, you can use :ref:`inflection-configuration` to
 customize the rules used::
 
     // Apples
-    echo Inflector::pluralize('Apple');
+    echo Inflector::pluralize('apple');
```
Comment (markstory) [inline en/core-libraries/inflector.rst:72]: The comment above is now incorrect with this change. The output would be `apples` with a lowercased a.

File: en/core-libraries/inflector.rst [hunk 2]
```diff
@@ -72,20 +69,24 @@ to support other languages, you can use :ref:`inflection-configuration` to
...
-    ``pluralize()`` may not always correctly convert a noun that is already in its plural form.
+    ``pluralize()`` should not be used on a noun that is already in its plural form.
+    For correct results with multi-words you should not use CamelCase versions. Transform
+    them to under\_scored first.
 
 .. code-block:: php
 
     // Person
-    echo Inflector::singularize('People');
+    echo Inflector::singularize('people');
```
Comment (markstory) [inline en/core-libraries/inflector.rst:83]: Same here with `person` instead of `Person`.

**Comment by dereuromark** [comment]
But given that we use CamelCase versions also directly in code

    $table = Inflector::pluralize($entityClass);

I think we might want to do the fix up for this case instead (the core PR).

**Comment by dereuromark** [comment]
Closing in favor of https://github.com/cakephp/docs/pull/6664

**Pull request closed without merging**